src/plugins/data/public/search/search_service.ts

```diff
@@ -51,6 +53,9 @@ declare module 'kibana/public' {
 
 export interface ISearchStart {
   search: ISearchGeneric;
+  __LEGACY: {
+    esClient: any;
```

Comment:
Does this need to be `any`? I think we have a type for this (`APICaller`).

Reply:
I have a follow up PR doing that https://github.com/elastic/kibana/pull/54337/files#diff-c55654fe5ffdab0adcd2c5bda2f863e0R52